### PR TITLE
Allow UI configuration of SonarQube parameters in Gradle build.

### PR DESCRIPTION
**Pull request by HSAR**

### CODE REVIEW

File: Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -31,5 +32,23 @@
...
+  "loc.input.label.sqDbPassword": "Db User Password",
+  "loc.input.help.sqDbPassword": "SonarQube server 5.1 and lower only. Enter the password for the database user i.e. sonar.jdbc.password",
+  "loc.input.label.sqProjectName": "SonarQube Project Name",
+  "loc.input.help.sqProjectName": "The SonarQube project name, i.e. sonar.projectName. If not set, the plugin will select its own default. See: http://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner+for+Gradle",
+  "loc.input.label.sqProjectKey": "SonarQube Project Key",
+  "loc.input.help.sqProjectKey": "The SonarQube project unique key, i.e. sonar.projectKey. If not set, the plugin will select its own default. See: http://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner+for+Gradle",
+  "loc.input.label.sqProjectVersion": "SonarQube Project Version",
+  "loc.input.help.sqProjectVersion": "The SonarQube project version, i.e. sonar.projectVersion. If not set, the plugin will select its own default. See: http://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner+for+Gradle"
```
Comment (HSAR) [inline Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson]: Previous commit failed to add these localisation strings.

File: Tasks/SonarQubePreBuild/Strings/resources.resjson/en-US/resources.resjson [hunk 2]
```diff
@@ -10,7 +10,7 @@
   "loc.input.label.projectKey": "Project Key",
   "loc.input.help.projectKey": "The SonarQube project unique key, i.e. sonar.projectKey",
   "loc.input.label.projectName": "Project Name",
-  "loc.input.help.projectName": "The SonarQube project name, i.e. sonar.ProjectName",
+  "loc.input.help.projectName": "The SonarQube project name, i.e. sonar.projectName",
```
Comment (HSAR) [inline Tasks/SonarQubePreBuild/Strings/resources.resjson/en-US/resources.resjson]: Mistaken capitalisation, thought I'd fix it while I'm in the file.

File: Tasks/Common/sonarqube-common/sonarqube-common.ts [hunk 3]
```diff
@@ -3,12 +3,13 @@
 import tl = require('vsts-task-lib/task');
 import {ToolRunner} from 'vsts-task-lib/toolrunner';
 
-// Creates the tool runner for executing SonarQube.
-export function applySonarQubeParams(toolRunner:ToolRunner, sqHostUrl, sqHostUsername, sqHostPassword, sqDbUrl?, sqDbUsername?, sqDbPassword?):ToolRunner {
-
+// Applies required parameters for connecting a Java-based plugin to SonarQube.
+// sqDbUrl, sqDbUsername and sqDbPassword are required if the SonarQube version is less than 5.2.
+export function applySonarQubeConnectionParams(toolRunner:ToolRunner, sqHostUrl, sqHostUsername, sqHostPassword, sqDbUrl?, sqDbUsername?, sqDbPassword?):ToolRunner {
```
Comment (bgavrilMS) [inline Tasks/Common/sonarqube-common/sonarqube-common.ts]: Please add a remark to these functions stating that these arguments work for Maven and Gradle

**Comment by bgavrilMS** [comment]
LGTM

**Pull request merged**